pkg/controller/masteruserrecord/sync_test.go

```diff
@@ -159,24 +149,45 @@ func TestSynchronizeSpec(t *testing.T) {
 
 func TestSynchronizeStatus(t *testing.T) {
 	// given
-	logf.SetLogger(logf.ZapLogger(true))
-	scheme := apiScheme(t)
-
+	apiScheme(t)
 	mur := murtest.NewMasterUserRecord(t, "john",
 		murtest.StatusCondition(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")))
 
 	userAccount := uatest.NewUserAccountFromMur(mur,
 		uatest.StatusCondition(toBeNotReady("Provisioning", "")), uatest.ResourceVersion("123abc"))
 
-	// when and then
-	testSyncMurStatusWithUserAccountStatus(t, userAccount, mur, scheme, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
+	t.Run("successful", func(t *testing.T) {
+		// given
+		hostClient := test.NewFakeClient(t, mur)
+		sync, memberClient := prepareSynchronizer(t, userAccount, mur, hostClient)
+
+		// when
+		err := sync.synchronizeStatus()
+
+		// then
+		require.NoError(t, err)
+		verifySyncMurStatusWithUserAccountStatus(t, memberClient, hostClient, userAccount, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
+	})
+
+	t.Run("failed on the host side", func(t *testing.T) {
+		// given
+		hostClient := test.NewFakeClient(t, mur)
+		hostClient.MockStatusUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+			return fmt.Errorf("some error")
+		}
+		sync, _ := prepareSynchronizer(t, userAccount, mur, hostClient)
+
+		// when
+		err := sync.synchronizeStatus()
+
+		// then
+		require.Error(t, err)
```

Comment:
maybe we should also check the error itself?

Reply:
added 1a7b7de